cloudevents/spec.md

```diff
@@ -118,7 +118,7 @@ information.
 #### Event Format
 
 An Event Format specifies how to serialize a CloudEvent as a sequence of bytes.
-Stand-alone event formats, such as the [JSON format](formats/json-format.md),
+Structured event formats, such as the [JSON format](formats/json-format.md),
```

Comment:
The [terminology section](https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#event-format) of the spec talks about _structured mode_ vs. _binary mode_. The expression _stand-alone format_ is introduced for event formats that do not depend on a protocol:
> Stand-alone event formats, such as the [JSON format](https://github.com/cloudevents/spec/blob/main/cloudevents/formats/json-format.md), specify serialization independent of any protocol or storage medium. Protocol Bindings MAY define formats that are dependent on the protocol.

Reply:
Is this comment just an FYI, or are you saying we need to keep "stand-alone" here?

Reply:
I'm sorry, I actually didn't realize that you are intending to change exactly the part of the spec I was quoting. I thought you were just changing the primer and wanted to maintain consistency with the spec. 

Nevertheless, I think _stand-alone event format_ is not synonymous with _structured mode_. The term was introduced because of AMQP, where you could have a _structured format_ that would NOT be a _stand-alone format_. 

Reading through the terminology again, I get the feeling that it is not fully consistent regarding the distinction of event formats, messages, and binary vs. structured mode. I could explain in next week's call what I mean. We can then decide, if we extend this PR, or if we create a separate one.

Reply:
What's the diff between "structured" and "stand-alone" to you? If there really is a diff I think we need to define "stand-alone" and be more clear on when each is used because I think the spec/primer today use them interchangeably.

Reply:
In my opinion ANY usage of an event format suggests structured mode. We intentionally introduced structured vs. binary MODE. The addition _stand-alone_ was just introduced to distinguish formats that are independent from the used protocol from others that have dependencies. As an example for a non-stand-alone format you could think of an AMQP format. The AMQP messaging layer allows to specify an AMQP sequence or value instead of a binary section. In this case the whole event would be transformed to the AMQP type system. 
For the nesting case I intentionally used the term _stand-alone_, because the inner event is supposed to be in a protocol-independent format. 
I see an event format as a function _f(context, data)_ that returns a byte stream. In binary mode you don't really have that, as _context_ and _data_ are handled separately.

Reply:
I think I understand what you're saying but I think it's confusing to a reader because we use the term "stand-alone" as the generic way to talk about "structured" messages. And at an abstract level that might be accurate, but I think it adds confusion because in our cases "stand-alone" == "structured", so switching between the two (or switching between being "abstract" and "concrete") doesn't change the point in the text we're trying to make... which is, when they must use "a structured CE" or not.

In the case of a nested CE, the inner CE (I believe) MUST be a structured CE, not just a "stand-alone" message, right? If you're suggesting that for certain serializations their "stand-alone" format allows for CE attributes to be injected and therefore we don't need a separate CE structured format, then that might be true but I would think then that the stand-alone+CE attributes definition IS the CE structured format for that serialization. Which means we can still use the term "CE structure" instead of "stand-alone". But a I re-read what I wrote, maybe I'm still missing your point :-)  Let's chat off-line